application/service.go

```diff
@@ -91,12 +93,16 @@ func (s *ServiceImpl) IsValid(request *web.Request) bool {
 		return true
 	}
 
-	formToken, err := request.Form1(TokenName)
+	requestToken, err := request.Form1(FormTokenName)
 	if err != nil {
-		return false
+		headerCsrfToken := request.Request().Header.Get(HeaderTokenName)
```

Comment:
To be honest, I don't like this approach - it feels a little bit magical to put whole this logic into single method. Main idea of the method is to validate form POST parameter - so I would suggest to expose two methods, one for validating header parameter, second to validates form POST parameter.

Reply:
You‘re right. I‘ll separate these two checks from each other.